source/Calamari.Tests/Fixtures/StructuredVariables/StructuredConfigVariablesServiceFixture.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.StructuredVariables;
+using Calamari.Common.Plumbing.Deployment;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.StructuredVariables
+{
+    [TestFixture]
+    public class StructuredConfigVariablesServiceFixture
+    {
+        const string FileName = "config.json";
+        static readonly string AdditionalPath = Path.GetFullPath("/additionalPath/");
+        static readonly string CurrentPath = Path.GetFullPath("/currentPath/");
+        static readonly string ConfigFileInCurrentPath = Path.Combine(CurrentPath, FileName);
+        static readonly string ConfigFileInAdditionalPath = Path.Combine(AdditionalPath, FileName);
+        
+        void RunAdditionalPathsTest(
+            bool fileExistsInPath, 
+            bool fileExistsInAdditionalPath,
+            Action<IStructuredConfigVariableReplacer> replacerAssertions = null,
+            Action<InMemoryLog> logAssertions = null)
+        {
+            var fileSystem = Substitute.For<ICalamariFileSystem>();
+            fileSystem.FileExists(ConfigFileInCurrentPath).Returns(fileExistsInPath);
+            fileSystem.FileExists(ConfigFileInAdditionalPath).Returns(fileExistsInAdditionalPath);
+            fileSystem.EnumerateFilesWithGlob(CurrentPath, FileName)
+                      .Returns(fileExistsInPath ? new[]{ ConfigFileInCurrentPath } : new string[0]);
+            fileSystem.EnumerateFilesWithGlob(AdditionalPath, FileName)
+                      .Returns(fileExistsInAdditionalPath ? new[]{ ConfigFileInAdditionalPath } : new string[0]);
+            
+            var replacer = Substitute.For<IStructuredConfigVariableReplacer>();
+            var log = new InMemoryLog();
+            var service = new StructuredConfigVariablesService(replacer, fileSystem, log);
+
+            var variables = new CalamariVariables();
+            variables.Set(ActionVariables.AdditionalPaths, AdditionalPath);
+            variables.AddFlag(PackageVariables.JsonConfigurationVariablesEnabled, true);
+            variables.Set(PackageVariables.JsonConfigurationVariablesTargets, FileName);
+            variables.Set(PackageVariables.CustomInstallationDirectory, CurrentPath);
+
+            var deployment = new RunningDeployment(CurrentPath, variables)
+            {
+                CurrentDirectoryProvider = DeploymentWorkingDirectory.CustomDirectory
+            };
+
+            service.ReplaceVariables(deployment);
+
+            replacerAssertions?.Invoke(replacer);
+            logAssertions?.Invoke(log);
+        }
+
+        [Test]
+        public void ReplacesVariablesInAdditionalPathIfFileMatchedInWorkingDirectory()
+        {
+            void Assertions(IStructuredConfigVariableReplacer replacer)
```

Comment:
I need to pass an `Action<...>` to `RunAdditionalPathsTest` on line 68, but given our current code formatting rules, this is the simplest way I can do it without the code being indented way over to the right.